src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -9,19 +9,38 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     internal class ProvideSemanticTokensResponse
     {
-        public ProvideSemanticTokensResponse(SemanticTokens result, long? hostDocumentSyncVersion)
+        public ProvideSemanticTokensResponse(SemanticTokens? result, SemanticTokensDelta? editResult, long? hostDocumentSyncVersion)
         {
             Result = result;
+            EditResult = editResult;
             HostDocumentSyncVersion = hostDocumentSyncVersion;
         }
 
-        public SemanticTokens Result { get; }
+        public SemanticTokens? Result { get; }
+
+        public SemanticTokensDelta? EditResult { get; }
 
         public long? HostDocumentSyncVersion { get; }
 
-        public override bool Equals(object obj) =>
-            obj is ProvideSemanticTokensResponse other &&
-            other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion) && other.Result.Equals(Result);
+        public override bool Equals(object obj)
+        {
+            if (!(obj is ProvideSemanticTokensResponse other) || !other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion))
+            {
+                return false;
+            }
+
+            if (Result != null && other.Result != null && other.Result.Equals(Result))
```

Comment:
I believe this should be possible now:
```suggestion
            if (Result is not null && other.Result is not null && other.Result.Equals(Result))
```

Reply:
Why is `is not` better than `!=`?

Reply:
Both `== null` and `=! null` are technically overloadedable. Also IMO `is`/`is not` is just easier to parse visually.

https://www.gullberg.tk/blog/is-null-versus-null-in-c/